Engine/SubscriptionDataReader.cs

```diff
@@ -554,23 +558,29 @@ private SubscriptionStreamReader GetReader(string source)
                     // check if this is not a local uri then download it to the local cache
                     if (uri.IsAbsoluteUri && !uri.IsLoopback)
                     {
-                        try
+                        //Skip download if the file is already in cache
+                        if (!File.Exists(location))
```

Comment:
Here's a [link](https://lean.quantconnect.com/docs#topic15.html) to the Lean docs on static vs dynamic data sources.